apis/crd/v1alpha1/cloudprovideraccount_types.go

```diff
@@ -38,14 +38,14 @@ type CloudProviderAccountAWSConfig struct {
 	// Reference to k8s secret which has cloud provider credentials.
 	SecretRef *SecretReference `json:"secretRef,omitempty"`
 	// Cloud provider account region.
-	Region string `json:"region,omitempty"`
+	Region []string `json:"region,omitempty"`
 	// Endpoint URL that overrides the default AWS generated endpoint.
 	Endpoint string `json:"endpoint,omitempty"`
 }
 
 type CloudProviderAccountAzureConfig struct {
 	SecretRef *SecretReference `json:"secretRef,omitempty"`
-	Region    string           `json:"region,omitempty"`
+	Region    []string         `json:"region,omitempty"`
```

Comment:
I don't think we allow region field to be empty.
So, we have to remove 'omitempty' for the region field.

Reply:
Agree that region is a required field, removed the omitEmpty tag.
However, In webhook there is a check already to make sure region field exists, hence not sure removing the omitEmpty field is really required. I assume even SecretReference is mandatory, yet we have omitEmpty tag for it.